src/client.rs

```diff
@@ -1019,12 +1019,37 @@ where
                     None => {
                         trace!("Waiting for message inside transaction or in session mode");
 
-                        match tokio::time::timeout(
-                            idle_client_timeout_duration,
-                            read_message(&mut self.read),
-                        )
-                        .await
-                        {
+                        let message = tokio::select! {
```

Comment:
I tested this idea trying to see if this helps with https://github.com/postgresml/pgcat/issues/303

Something else it's missing that doesn't allow to send the server messages to the client. Maybe this 2 issues are related and could be solved both with the same fix